Log panel: Allow to filter by name

achim-k opened the pull request:
**User-Facing Changes**
Log panel: Allow to filter by name

**Description**
Allows to filter log messages by publisher name

![image](https://github.com/foxglove/studio/assets/9250155/b898be50-a3fb-4cca-bc83-839f2bd0c1b3)

Resolves FG-4172

snosenzo commented on packages/studio-base/src/panels/Log/index.tsx:
```diff
@@ -113,11 +94,67 @@ const LogPanel = React.memo(({ config, saveConfig }: Props) => {
+    nameFilter,
+    updatePanelSettingsTree,
+    seenNodeNames,
+    seenNodeNames.size, // Needed as seenNodeNames doesn't
```
unfinished comment

snosenzo reviewed (approved):
The behavior of accumulating the node names is a little strange, but I'm happy to see it remembers their status after reloading. 
Wondering if we also want the Log Settings panel to have a search-for-settings filter (like the 3D panel) at the top in case the name list is really long?
<img width="802" alt="image" src="https://github.com/foxglove/studio/assets/10187776/29785b7d-6300-4d05-8e06-63c19939a618">

Otherwise I think it looks good, but it sounds like design/other folks might also want a pass

snosenzo commented on packages/studio-base/src/panels/Log/filterMessages.test.ts:
```diff
@@ -121,8 +129,19 @@ describe("filter", () => {
+        filterMessages(ros1msgs, {
+          minLogLevel,
+          searchTerms: [],
+          nameFilter: { "/some_topic": { visible: false } },
```
another test with visible:true?

amacneil commented:
LR feedback:
- Drop node autocomplete from filter box

Feedback on filter:
- Move log level filter dropdown to sidebar
- Move search filter to top of panel (below panel titlebar)
- Remove copy button

amacneil commented on packages/studio-base/src/i18n/en/log.ts:
```diff
@@ -5,4 +5,7 @@
 export const log = {
   topicError: "Topic {{topic}} is not available",
   topic: "Topic",
+  nameFilter: "Name filter",
```
```suggestion
  nameFilter: "Namespaces",
```

achim-k commented:
Addressed all comments, this one is ready for another review

![image](https://github.com/foxglove/studio/assets/9250155/76296171-6ccc-410c-aa21-aad62903ecb8)

snosenzo commented on packages/studio-base/src/panels/Log/FilterBar.tsx:
```diff
@@ -53,85 +19,21 @@ type Filter = {
-        </ToolbarIconButton>
-      </Stack>
-    </Stack>
+    <FilterTagInput
```
now that its only this, would it make more sense to pull this out and use this directly in the parent container?

achim-k replied:
Yes, I can do that

snosenzo reviewed (approved):
Tested, looks good to me 👍

jtbandes commented on packages/studio-base/src/panels/Log/index.stories.tsx:
```diff
@@ -319,3 +319,22 @@ export const FoxgloveLog: StoryObj = {
+
+export const FilterByName: StoryObj = {
+  render: () => (
+    <PanelSetup fixture={fixture}>
```
It would be nice to show what the settings tree looks like in these stories too (with some filters enabled/disabled). You can add `includeSettings` to the PanelSetup

jtbandes reviewed (approved):
This story seems either broken or no longer relevant:
<img width="1560" alt="image" src="https://github.com/foxglove/studio/assets/14237/37621526-3716-4a3c-a6ef-818e80da4b64">
